scanpy/experimental/pp/_highly_variable_genes.py

```diff
@@ -105,24 +162,41 @@ def _highly_variable_pearson_residuals(
         if clip < 0:
             raise ValueError("Pearson residuals require `clip>=0` or `clip=None`.")
 
+        residual_gene_var = np.zeros((X_batch.shape[1]), dtype=np.float64)
```

Comment:
Please avoid mutating parameters. You can allocate this from within the functions, no?

Reply:
I think you are right with this. I was still thinking in cuda kernel terms and there you have to create the array outside of the function